src/main/kotlin/ar/edu/unahur/obj2/semillasAlViento/Parcela.kt

```diff
@@ -1,35 +1,35 @@
 package ar.edu.unahur.obj2.semillasAlViento
 
 class Parcela(val ancho: Int, val largo: Int, val horasSolPorDia: Int) {
+
   val plantas = mutableListOf<Planta>()
-  var cantidadPlantas = 0
+
+  // Se elimina la variable cantidadPlantas por redundancia
 
   fun superficie() = ancho * largo
+
+  //En vez de calcular ancho * largo, usa método superficie()
   fun cantidadMaximaPlantas() =
-    if (ancho > largo) ancho * largo / 5 else ancho * largo / 3 + largo
+    if (ancho > largo) superficie()/ 5 else superficie() / 3 + largo
+
 
   fun plantar(planta: Planta) {
-    if (cantidadPlantas == this.cantidadMaximaPlantas()) {
-      println("Ya no hay lugar en esta parcela")
+    // Reemplazamos el string por un error
+    if (plantas.size == this.cantidadMaximaPlantas()) {
+      throw error ("Ya no hay lugar en esta parcela")
     } else if (horasSolPorDia > planta.horasDeSolQueTolera() + 2) {
-      println("No se puede plantar esto acá, se va a quemar")
+      throw error ("No se puede plantar esto acá, se va a quemar")
     } else {
       plantas.add(planta)
-      cantidadPlantas += 1
     }
   }
+  fun tieneComplicaciones() =
+    plantas.any { it.horasDeSolQueTolera() < horasSolPorDia }
 }
 
-class Agricultora(val parcelas: MutableList<Parcela>) {
-  var ahorrosEnPesos = 20000
-
-  // Suponemos que una parcela vale 5000 pesos
-  fun comprarParcela(parcela: Parcela) {
-    if (ahorrosEnPesos >= 5000) {
-      parcelas.add(parcela)
-      ahorrosEnPesos -= 5000
-    }
-  }
+class Agricultora(val parcelas: List<Parcela>) {
+  // Se elimina la variable ahorroEnPesos y se elimina el metodo comprarParcela
+  // val parcela ahora es inmutable
 
   fun parcelasSemilleras() =
     parcelas.filter {
```

Comment:
La responsabilidad de saber si una Parcela es semillera o no debería ser de la misma Parcela.
```suggestion
  fun parcelasSemilleras() = parcelas.filter { parcela -> parcela.esSemillera() }
```